src/main/javascript/crypto/src/eas/EasTicketAttestation.ts

```diff
@@ -449,35 +467,45 @@ export class EasTicketAttestation extends AttestableObject implements Attestable
 			parts.domain
 		);
 
-		const domain: EIP712DomainTypedData = {
-			name: "EAS Attestation",
-			version: domainDecoded[0],
-			verifyingContract: domainDecoded[1],
-			chainId: domainDecoded[2]
-		}
-
 		// console.log("ABI Encoded bytes: ", "0x" + uint8tohex(new Uint8Array(asnEmbedded.easAttestation)));
 
+		const firstElem = defaultAbiCoder.decode(['uint16'], parts.attestation);
+		const version = firstElem[0] > 50 ? 0 : 1;
```

Comment:
@micwallace , why do you use 50 as limit? in current implementation we allow only version 1,
so to check more bit and avoid collision when schema last bytes are 0x0001 , better use
```
const firstElem = defaultAbiCoder.decode(['uint256'], parts.attestation);
const version = firstElem[0] > 50n ? 0 : 1;
```
it will cover more bytes

Reply:
Honesty I didn't know how to read the length of the ABI element, so I just went with a safe default lol.
I've applied this change it's working well.